docs/csharp/programming-guide/classes-and-structs/destructors.md

```diff
@@ -1,85 +1,83 @@
 ---
 title: "Finalizers - C# Programming Guide"
-description: Finalizers in C#, which are also called destructors, perform any necessary final clean-up when a class instance is being collected by the garbage collector.
-ms.date: 10/08/2018
+description: Finalizers in C# perform any necessary final clean-up when a class instance is being collected by the garbage collector.
+ms.date: 06/09/2021
 helpviewer_keywords: 
   - "~ [C#], in finalizers"
   - "C# language, finalizers"
   - "finalizers [C#]"
-ms.assetid: 1ae6e46d-a4b1-4a49-abe5-b97f53d9e049
 ---
 # Finalizers (C# Programming Guide)
 
-Finalizers (which are also called **destructors**) are used to perform any necessary final clean-up when a class instance is being collected by the garbage collector.  
-  
+Finalizers (historically referred to as **destructors**) are used to perform any necessary final clean-up when a class instance is being collected by the garbage collector.
+
+In most cases, you can avoid writing a finalizer by using the  <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=fullName> or derived classes to wrap any unmanaged handle.
```

Comment:
The intro isn't a paragraph, it's two separate sentences. Do you think it makes sense to converge this a bit? Also, since we're calling out `SafeHandle` as an alternative approach, does it make sense to include that as an example?

Reply:
I didn't add an example, because those explode quickly. `SafeHandle` has several derived classes for Files, network pipes, memory handles, and more. Depending on which kind of resource you're holding, you'd use different derived classes.

I think pointing readers to that class will help them find the correct derived class for their use.